### PR TITLE
fix: fix bytes abi transformation

### PR DESCRIPTION
**Pull request by Kuzirashi**
Related to: https://github.com/nervosnetwork/polyjuice-provider/issues/88

### CODE REVIEW

**Comment by RetricSu** [comment]
please fix ci @Kuzirashi

**Comment by Kuzirashi** [comment]
Fixed @RetricSu

File: packages/base/src/abi.ts [hunk 1]
```diff
@@ -257,7 +257,7 @@ export function decodeInputDataByAbi(data: HexString, abiItem: AbiItem) {
 
     retData.params.push({
       name: abiItem.inputs[i].name,
-      value: parsedParam,
+      value: parsedParam === null ? "0x" : parsedParam,
```
Comment (RetricSu) [inline packages/base/src/abi.ts:260]: I give it some thoughts. the problem is that one of provider dependencies library `web3-eth-abi` treat `0x` for type `bytes` as null when decoding parameters https://github.com/nervosnetwork/polyjuice-provider/blob/main/packages/base/src/abi.ts#L221-L224 , 
and the problem only happens to bytes, not even bytes2 / bytes8 / ... bytes32.

simply add `parsedParam === null ? "0x" : parsedParam` might not be the best choice.  instead, we can add `const isBytes = abiItem.inputs[i].type.indexOf("bytes") === 0; ` here https://github.com/nervosnetwork/polyjuice-provider/blob/main/packages/base/src/abi.ts#L235, and check if bytes type value is null, if so, replace with 0x

**Review by RetricSu — CHANGES_REQUESTED** [review]
I give it some thoughts. the problem is that one of provider dependencies library `web3-eth-abi` treat `0x` for type `bytes` as `null` when [decoding parameters](https://github.com/nervosnetwork/polyjuice-provider/blob/main/packages/base/src/abi.ts#L221-L224) , and the problem only happens to bytes, not even bytes2 / bytes8 / ... bytes32.

simply add `parsedParam === null ? "0x" : parsedParam` might not be the best choice.  instead, we can add `const isBytes = abiItem.inputs[i].type.indexOf("bytes") === 0; ` [here](https://github.com/nervosnetwork/polyjuice-provider/blob/main/packages/base/src/abi.ts#L235), and check if bytes type value is null, if so, replace with 0x

**Comment by Kuzirashi** [comment]
Can you check again @RetricSu ?

**Review by RetricSu — APPROVED** [review]

**Pull request merged**